### PR TITLE
Speed up tests / fix failure hang.

### PR DESCRIPTION
**Pull request by theengineear**
Attempts to speed up overall test running in GitHub actions and fixes an edge case where we wait the full timeout before exiting a failed test.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "start": "node ./server.js",
     "lint": "eslint --max-warnings=0 .",
     "lint-fix": "eslint --fix .",
-    "ci:setup-playwright": "playwright install --with-deps chromium",
```
Comment (theengineear) [inline package.json:22]: This is slow — hopefully the environment in GH actions doesn’t _actually_ need it.

File: x-test-cli-tap.js [hunk 2]
```diff
@@ -195,15 +197,20 @@ export class XTestCliTap {
...
+    const sets = this.#state.inYaml
+      ? [XTestCliTap.#inYamlPatterns]
       : this.#state.inFailureBlock
-        ? XTestCliTap.#inFailurePatterns
-        : XTestCliTap.#patterns;
-    for (const pattern of Object.values(patterns)) {
-      const match = pattern.exec(line);
-      if (match) {
-        return { pattern, match };
+        ? [XTestCliTap.#inFailurePatterns, XTestCliTap.#patterns]
```
Comment (theengineear) [inline x-test-cli-tap.js:206]: The TAP plan comes after the failures block, we just need to look for it.
